node/src/service.rs

```diff
@@ -636,51 +660,56 @@ fn build_manual_seal_import_queue(
     ))
 }
 
-fn build_consensus_container(
+async fn start_consensus_container(
```

Comment:
Yes but this async is needed if we want to keep calling the `warp` function with `start_consensus_container` inside it. This is because there is a constraint in the warp fn in which the function that we send via parameter must implement `Future`. We could also remove that constraint, but I don't know how that would affect logs, what do you think?

Reply:
I see, let's add a TODO then. All the `collate_on` stuff was async in the past but does not need to be async anymore. But changing that will be quite a big refactor, so we can do it later. The logs issue can be fixed adding the attribute here as well, this way we can safely remove the wrap function in the future:

```suggestion
// TODO: this function does not need to be async
#[sc_tracing::logging::prefix_logs_with(container_log_str(para_id))]
async fn start_consensus_container(
```